server/modules/rendering/html-security/renderer.js

```diff
@@ -32,6 +32,59 @@ module.exports = {
         allowedAttrs.push('allow')
       }
 
+      //Changes to keep interactive plantuml object tag
```

Comment:
Should be an option, set to Off by default, as this opens up potential security issues.

Reply:
Hi @NGPixel 

Currently, the allowedAttrs/allowedTags is being used to trigger logic in the uponSanitizeElement logic to remove the node completely if it doesn't meet the criteria for plantuml.

This logic validates that the url is set to the preconfigured plantuml server and overwrites atrribute values being passed to acceptable values, else rejects the node altogether

I tried to see if there was a hook that let me parse the content and adjust it without setting the `allowAttributes` `allowTag` (e.g. `beforeSanitizeElement` however in that scenario, DOMpurify would always just remove the node.

So I had a read of use cases here: https://github.com/cure53/DOMPurify/blob/main/demos/README.md#hook-to-allow-and-sand-box-all-javascript-link

It does seem to suggest that we first allow the attributes and then use logic in uponSanitizeElement to remove them if they don't meet the criteria - this is what I have done.

Note that we do need to use an object node to make sure that the svg interactive i.e. to be select and copy text from the diagram or to be able to click links, the above steps ensure that the tag is secure by only allowing access to the preconfigured plantuml server and forcing other attribute values.

Detailed logic is
- Examine  all the plantuml tags and attributes;  To this I need to use `allowedTags` and `allowedAttrs` 
- In `uponSanitizeElement` - the node is rejected if it doesn't satisfy the criteria of a valid plantuml node i.e.
 - if the number of attributes doesn't match
 - if the url doesn't point to the configured plantuml server in wikis settings
- once the node is determined to be a plantuml node, the allowed attributes (as mentioned above) are forcefully overwritten to predefined values to make sure that no code injection is possible
- If no match is found, then remove the node

Reply:
@NGPixel  see changes in  commit [571d1ddf0e2ead23ee1cb7691555fa7e1a81bd13](https://github.com/Requarks/wiki/pull/2754/commits/571d1ddf0e2ead23ee1cb7691555fa7e1a81bd13) 

Summary:
- Object tag and attributes are allowed only if plantuml `svg` mode is selected. User already has the option to select png as an alternative so this is optional.
- To ensure that the user doesn't lose out too much by selecting png as the format - the plantuml content is now stored in hidden text in the html so that it is still searchable irrespective of the mode that the user selects (png or svg). However obviously user loses the interactiveness that he gets from selecting svg as the format (e.g. links/copy paste text).

Reply:
@NGPixel  just bumping this as a follow up - I believe I have incorporated your review comments, so please shout if you need any additional changes